nautilus/include/nautilus/val.hpp

```diff
@@ -110,6 +111,18 @@ class val<ValueType> {
 		return temp;
 	}
 
+	auto operator+() {
+		if constexpr (std::integral<ValueType> && sizeof(ValueType) < sizeof(int)) {
+			return static_cast<val<int>>(*this);
+		}
+		return *this;
+	}
+
+	val<ValueType> operator-() {
+		// negate (?) TODO i fear cppref has slightly different semantic
```

Comment:
please add a comment and describe the difference shortly

Reply:
I checked again and convinced myself that `0 - x` is the same as `-x`.